tests/test_trainable_tokens.py

```diff
@@ -619,3 +787,78 @@ def test_original_module_not_in_state_dict(self, model):
 
         state_dict = peft_model.state_dict()
         assert not [k for k in state_dict if ".original_module.weight" in k]
+
+    @pytest.fixture
+    def model_emb(self):
+        return ModelEmb()
+
+    @pytest.fixture
+    def model_embed_in(self):
+        return ModelEmbedIn()
+
+    @pytest.fixture
+    def model_embed_in_no_get(self):
+        return ModelEmbedInNoGet()
+
+    @pytest.fixture
+    def model_embed_multiple(self):
+        return ModelEmbedMultiple()
+
+    @pytest.mark.parametrize(
+        "model_fixture_name, getter",
+        [
+            ("model_emb", lambda model: model.emb),
+            ("model_embed_in", lambda model: model.embed_in),
+            ("model", lambda model: model.model.model.embed_tokens),
```

Comment:
It could be more prudent to use `operator.attrgetter` than `lambda` but maybe it's unproblematic here.

Reply:
In hindsight yes, I didn't know if there are models where it is a bit more complicated so I left it as lambdas. I can change it if you want.

Reply:
I know that `lambda` has some weird scoping rules, but I can never remember what exactly it was. If it's not relevant here, it's okay to leave it as is.